symfony/src/Service/MeasurementService.php

```diff
@@ -96,7 +96,16 @@ public function getMeasurmeent($participantId, $measurementId)
 
     public function requireBloodDonorCheck()
     {
-        return $this->params->has('feature.blooddonorpm') && $this->params->get('feature.blooddonorpm') && $this->session->get('siteType') === 'dv' && $this->siteService->isDiversionPouchSite();
+        return $this->params->has('feature.blooddonorpm') && $this->params->get('feature.blooddonorpm') && $this->session->get('siteType') === 'dv' && ($this->siteService->isDiversionPouchSite() || $this->isBloodDonorPmSite());
+    }
+
+    public function isBloodDonorPmSite(): bool
+    {
+        if ($this->params->has('feature.blooddonorpmsites') && $sites = $this->params->get('feature.blooddonorpmsites')) {
+            $sites = explode(',', $sites);
+            return in_array($this->siteService->getSiteId(), $sites);
+        }
+        return false;
```

Comment:
Did you consider putting this method in the site service class? The diversion pouch site check seems similar, and might make sense to organize together.